tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
```

Comment:
Also, I'm not sure if just counting events will work for many of the existing waitsFor() use cases.  I suspect you'll need to be able to provide a closure that inspects the event or similar.